phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -443,17 +448,44 @@ private Job configureJobForAysncIndex(String schemaName, String indexTable, Stri
 
                 // set input for map reduce job using hbase snapshots
                 PhoenixMapReduceUtil
-                    .setInput(job, PhoenixIndexDBWritable.class, snapshotName, qDataTable, restoreDir, selectQuery);
+                            .setInput(job, PhoenixIndexDBWritable.class, snapshotName, qDataTable, restoreDir, selectQuery);
             }
             TableMapReduceUtil.initCredentials(job);
             
             if (useDirectApi) {
-                return configureSubmittableJobUsingDirectApi(job, false);
+                job.setMapperClass(PhoenixIndexImportDirectMapper.class);
+                return configureSubmittableJobUsingDirectApi(job);
             } else {
                 return configureRunnableJobUsingBulkLoad(job, outputPath);
-                
             }
-            
+        }
+
+        private Job configureJobForServerBuildIndex()
+                throws Exception {
+
+            PhoenixConfigurationUtil.setIndexToolDataTableName(configuration, qDataTable);
+            PhoenixConfigurationUtil.setIndexToolIndexTableName(configuration, qIndexTable);
+
+            String physicalIndexTable = pIndexTable.getPhysicalName().getString();
+
+            PhoenixConfigurationUtil.setPhysicalTableName(configuration, physicalIndexTable);
+            PhoenixConfigurationUtil.setDisableIndexes(configuration, indexTable);
+
+            fs = outputPath.getFileSystem(configuration);
+            fs.delete(outputPath, true);
+
+            final String jobName = String.format(INDEX_JOB_NAME_TEMPLATE, schemaName, dataTable, indexTable);
+            final Job job = Job.getInstance(configuration, jobName);
+            job.setJarByClass(IndexTool.class);
+            job.setMapOutputKeyClass(ImmutableBytesWritable.class);
+            FileOutputFormat.setOutputPath(job, outputPath);
```

Comment:
Can dispense with the output path, I think

Reply:
When I remove the code for setting the output path, there is always one other test failing. There are total of 120 tests. I tried restructuring the IndexTool code so that the output path is set only when bulk loading is enabled but did not help. So, I will leave the code as it is.